### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by Radhikajram**
Week3 Changes

### CODE REVIEW

File: homework/App.js [hunk 1]
```diff
@@ -16,14 +16,78 @@ class App {
     // 1. Create the fixed HTML elements of your page
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
 
+    const BodyEl = document.body;
+
     const root = document.getElementById('root');
 
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    // Create div element 'select' in  document body to hold the label element and list box.
```
Comment (grod220) [inline homework/App.js:23]: Really solid comments 👍

File: homework/App.js [hunk 2]
```diff
@@ -16,14 +16,78 @@ class App {
...
+
+      sortRepoName.forEach(loadrepo => {
+        const option = document.createElement('option');
+        option.value = loadrepo;
+        option.text = loadrepo;
+        selectBox.appendChild(option);
+      });
+
+      let keyIndex;
+      console.log(selectBox.value);
```
Comment (grod220) [inline homework/App.js]: In general, best to remove console.logs before merging. So your other colleagues don't get annoyed with your logs when they work in different parts of the app ;)

File: homework/App.js [hunk 3]
```diff
@@ -16,14 +16,78 @@ class App {
...
+
+      selectBox.addEventListener('change', () => {
+        Object.keys(repos).forEach(key => {
+          if (selectBox.value === repos[key].name) {
+            keyIndex = key;
+          }
+        });
+        this.fetchContributorsAndRender(keyIndex);
+      });
     } catch (error) {
```
Comment (grod220) [inline homework/App.js:90]: This catch could probably just wrap the `const repos = await Util.fetchJSON(url);` line. Being picky though.

**Review by grod220 — APPROVED** [review]
Wow! 🎆 
This assignment was definitely a toughie. Well done ✅

File: homework/App.js (hunk 2 above)
Comment (Radhikajram) [reply homework/App.js]: I have remove the console.log statement.

Comment (Radhikajram) [reply homework/App.js]: Shall i merge the Week3 branch with master?

Comment (grod220) [reply homework/App.js]: Approved for merging 👍

**Pull request merged**